src/csync/csync_update.cpp

```diff
@@ -712,7 +713,30 @@ int csync_ftw(CSYNC *ctx, const char *uri, csync_walker_fn fn,
       goto error;
   }
 
-  while ((dirent = csync_vio_readdir(ctx, dh))) {
+  while (true) {
+    errno = 0;
+    dirent = csync_vio_readdir(ctx, dh);
+    if (!dirent) {
+        // Normal case: End of items in directory
+        if (errno == 0)
+            break;
+
+        // Note: Windows vio converts any error into EACCES
+        qCWarning(lcUpdate, "readdir failed for file in %s - errno %d", uri, errno);
+
+        // If we get an error during readdir, maybe that's a single unreadable
+        // item that we can skip? It's unclear whether that can happen.
+        // We definitely don't want to get stuck in an infinite loop here,
+        // so stop if we get the same errno twice in a row.
+        if (errno == last_errno) {
+            qCWarning(lcUpdate, "Second failure, giving up on directory");
+            goto error;
+        }
+        last_errno = errno;
+        continue;
```

Comment:
I'm not sure if we should continue. 
If it is a temporary error, for example, we might end up skipping files in the discovery, meaning that the client would believe they were removed, so it would delete them on the server.
Aborting the sync seems a safer option,  or at least somehow ignoring any removed item in this directory.